watertap/unit_models/anaerobic_digestor.py

```diff
@@ -0,0 +1,880 @@
+            units=pyunits.m**3 / pyunits.day / pyunits.bar,
+            doc="friction parameter",
+        )
+
+        self.KH_co2 = Param(
+            initialize=0.02715,
+            units=pyunits.kmol / pyunits.m**3 * pyunits.bar**-1,
+            mutable=True,
+            doc="CO2 Henry's law coefficient",
+        )
+        self.KH_ch4 = Param(
+            initialize=0.00116,
+            units=pyunits.kmol / pyunits.m**3 * pyunits.bar**-1,
+            mutable=True,
+            doc="CH4 Henry's law coefficient",
+        )
+        self.KH_h2 = Param(
+            initialize=7.38e-4,
+            units=pyunits.kmol / pyunits.m**3 * pyunits.bar**-1,
+            mutable=True,
+            doc="H2 Henry's law coefficient",
+        )
+        self.K_La = Param(
+            initialize=200,
+            units=pyunits.day**-1,
+            mutable=True,
+            doc="Gas-liquid transfer coefficient",
+        )
+
+        def outlet_P_rule(self, t):
+            return self.vapor_phase[t].pressure == (
+                self.vapor_phase[t].p_w_sat
+                + sum(
+                    self.vapor_phase[t].p_sat[j]
+                    for j in self.config.vapor_property_package.solute_set
+                )
+            )
+
+        self.outlet_P = Constraint(
+            self.flowsheet().time,
+            rule=outlet_P_rule,
+            doc="Outlet vapor phase pressure",
+        )
+
+        # Material balances
+        def rule_material_balance(self, t, j):
+            if j in common_comps:
+                # Component is in equilibrium
+                # Mass transfer equals vapor flowrate
+                return self.liquid_phase.mass_transfer_term[
+                    t, "Liq", j
+                ] == -self.vapor_phase[t].get_material_flow_terms("Vap", j)
+
+            elif j == "S_IC":
+                # Mass transfer term is zero, no vapor flowrate
+                return self.liquid_phase.mass_transfer_term[
+                    t, "Liq", j
+                ] == -self.vapor_phase[t].get_material_flow_terms("Vap", "S_co2")
+
+            elif j in self.liquid_phase.properties_out.component_list:
+                # No mass transfer term
+                # Set vapor flowrate to an arbitary small value
+                return self.liquid_phase.mass_transfer_term[
+                    t, "Liq", j
+                ] == 1e-8 * lunits(fb)
+
+        self.unit_material_balance = Constraint(
+            self.flowsheet().time,
+            self.liquid_phase.properties_out.component_list,
+            rule=rule_material_balance,
+            doc="Unit level material balances",
+        )
+
+        def Sh2_conc_rule(self, t):
+            return self.liquid_phase.mass_transfer_term[t, "Liq", "S_h2"] == -1 * (
+                pyunits.convert(self.K_La, to_units=1 / pyunits.s)
+                * (
+                    self.liquid_phase.properties_out[t].conc_mass_comp["S_h2"]
+                    - 16
```

Comment:
Unsure of where the value of 16 comes from. Similarly, the value of 64 in the next constraint. I realize you're converting via molecular weight here but maybe also multiplying by mole fractions based on stoichiometry and curious to clarify. Might also be nice to include comments for further clarification. #909